flow/flat_buffers.cpp

```diff
@@ -563,4 +573,54 @@ TEST_CASE("/flow/FlatBuffers/EmptyPreSerVectorRefs") {
 	return Void();
 }
 
+TEST_CASE("/flow/FlatBuffers/EmptyUnorderedSet") {
+	int kSize = deterministicRandom()->randomInt(0, 100);
+	Standalone<StringRef> msg =
+	    ObjectWriter::toValue(std::vector<std::unordered_set<Y1, Y1Hasher, Y1Equal>>(kSize), Unversioned());
+	ObjectReader rd(msg.begin(), Unversioned());
+	std::vector<std::unordered_set<Y1, Y1Hasher, Y1Equal>> xs;
+	rd.deserialize(xs);
+	ASSERT(xs.size() == kSize);
+	for (const auto& x : xs) {
+		ASSERT(x.size() == 0);
+	}
+	return Void();
+}
+
+TEST_CASE("/flow/FlatBuffers/NonEmptyUnorderedSet") {
+	// first we construct the data to serialize/deserialize
+	// so we can compare it afterwards
+	std::unordered_set<std::string> src;
+	src.insert("FDB");
+	src.insert("is");
+	src.insert("awesome");
+	::Arena vecArena;
+	std::unordered_set<StringRef> outVec;
+	{
+		::Arena readerArena;
+		StringRef serializedVector;
+		{
+			::Arena arena;
+			std::unordered_set<StringRef> s;
+			for (const auto& str : src) {
+				s.insert(StringRef(arena, str));
+			}
+			ObjectWriter writer(Unversioned());
+			writer.serialize(FileIdentifierFor<decltype(s)>::value, arena, s);
+			serializedVector = StringRef(readerArena, writer.toStringRef());
+		}
+		ArenaObjectReader reader(readerArena, serializedVector, Unversioned());
+		// The VectorRef and Arena arguments are intentionally in a different order from the serialize call above.
```

Comment:
I don't think we need to duplicate testing the arena ownership/reordering stuff here - this test could be much simpler. You could just do something like you did with the empty set test above, with a non-empty `std::unordered_set<Y1, Y1Hasher, Y1Equal>`

Reply:
I think for the test to be extensive is not a bad idea; more testing is always better than lesser, IMHO. 

I don't have a strong opinion, but I would leave it as is unless you feel strong about it. 
Thanks

Reply:
Simulation provides the extensive testing. For the unit test I'm just looking for a test that says "if I serialize then deserialize some unordered_set, do I get the same set back?"

Reply:
Agree with you partially. Simulation is an amazing tool to validate the functionality for sure. 

However, if it is possible to validate more with unit-test, I find it cheap and easy to run option that we should explore; again, I don't means that all unit-test need not to be super complex and time consuming, which is the case here, imho.

Reply:
Updated the test to just validate "non-empty" unordered_set and avoid arena ownership/reordering verification. Thanks